x-pack/metricbeat/module/prometheus/remote_write/remote_write_test.go

```diff
@@ -18,6 +18,70 @@ import (
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
+func BenchmarkGenerateEvents(b *testing.B) {
```

Comment:
Can we consider writting  an integration test for the metrics_count enable?

You can create only one for the collector and add a comment that same stands for remote_write.
Idea would be to validate the metrics_count for one or two event families/ labels?

See eg from histograms: https://github.com/elastic/beats/blob/d732d5d5d211eb4bf39ce2b6f5689128dee016b8/x-pack/metricbeat/module/prometheus/collector/histogram_test.go

Reply:
@gizas Here's the sample output when I tested the change with actual Prometheus stats: https://github.com/elastic/beats/pull/40411#issue-2441684623 (See: **Sample output**)

Reply:
> Can we consider writting an integration test for the metrics_count enable as well?

Sure.

Reply:
Test added in this commit: https://github.com/elastic/beats/pull/40411/commits/3d93923d7f4d93356ea495ee3de4ac17c3db0f12

Reply:
(Sorry still have not run the tests locally, will try to do it and let u know)

Reply:
> My comment was more to compare those metrics with actual prometheus visualisations because our users this is what will do. They will take our metrics_count and will try to compare with an actual prometheus visualisation
>
> For eg. in remote_write we had (see old https://github.com/elastic/beats/issues/17079#issuecomment-612878489) we had rate(prometheus_remote_storage_samples_total[2m]) . Will this number be close to the sum (metrics_count) for period of 2m in our Kibana?
>
> I recall prometheus_tsdb_head_series shows the number of active time series.
>
> As long as you have a local setup I think a comparison would be benefitial even for us to answer following problems

Sure, will check this. Btw, I'll also DM you the doc where we have written why we want the field.

Reply:
I did test this yesterday. I am not able to see `prometheus_tsdb_head_series`. I checked Prometheus' Server UI and also checked if it is sending this to remote_write destination i.e., metricbeat; but no. But I did see `prometheus.scrape_samples_scraped.value` in Kibana and a few more. But nothing with *tsdb* or *remote_write*.